service/worker/scheduler/workflow.go

```diff
@@ -1180,8 +1176,10 @@ func (s *scheduler) startWorkflow(
 	nominalTimeSec := start.NominalTime.AsTime().UTC().Truncate(time.Second)
 	workflowID := newWorkflow.WorkflowId
 	if start.OverlapPolicy == enumspb.SCHEDULE_OVERLAP_POLICY_ALLOW_ALL || s.tweakables.AlwaysAppendTimestamp {
-		// must match AppendedTimestampForValidation
-		workflowID += "-" + nominalTimeSec.Format(time.RFC3339)
+		workflowID = InternalWorkflowIdRepresentation(
+			newWorkflow.WorkflowId,
+			nominalTimeSec,
+			s.StartScheduleArgs.GetSchedule().GetPolicies().GetKeepOriginalWorkflowId())
 	}
```

Comment:
If overlap policy is AllowAll, then we have to ignore the policy and always append the timestamp, otherwise AllowAll effectively becomes Skip.

Also, we should validate in Create+UpdateSchedule that KeepOriginalWorkflowId and AllowAll aren't used together. But even after we do that, we still have to handle it correctly here because we can override the overlap policy in trigger/backfill.